src/core/core-util.c

```diff
@@ -441,21 +442,23 @@ util_parse_time (const char *datetime, struct timeval *tv)
     if (pos_colon && !pos_hyphen)
     {
         /* add current date: "19:04:55" -> "2025-08-30T19:04:55" */
-        string = malloc (strlen (datetime) + 16 + 1);
+        string_size = strlen (datetime) + 16 + 1;
```

Comment:
I think the size here is still not good, it should be `strlen(str_date) + strlen(datetime) + 1`, according to what we put inside this string.

I'll take your changes and fix this.

Reply:
and bottom one should be `+ 9 + 1` instead of `+ 16 + 1`?

Reply:
Ah I was wrong, the str_date is less than 16 bytes in all cases.

So I think the initial code was not even buggy, but I understand the compiler may raise a warning due to `sizeof(str_date2)` being larger than the string.

Reply:
> `strlen(str_date) + strlen(datetime) + 1` makes tests fail here, not exactly sure why.

ah that's because i'm checking strlen too early.